pkg/install/kubeconfig.go

```diff
@@ -0,0 +1,93 @@
+package install
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"reflect"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/kubeconfig"
+	"github.com/openshift/installer/pkg/asset/tls"
+	clientcmd "k8s.io/client-go/tools/clientcmd/api/v1"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+)
+
+// generateAndStoreKubeconfigs generates additional admin credentials and kubeconfig based admin kubeconfig
+// found in graph and stores both to the database
+func (i *Installer) generateAndStoreKubeconfigs(ctx context.Context, g graph, aroServiceName string) error {
+
+	ca := g[reflect.TypeOf(&tls.AdminKubeConfigSignerCertKey{})].(*tls.AdminKubeConfigSignerCertKey)
+	clientCertKey := g[reflect.TypeOf(&tls.AdminKubeConfigClientCertKey{})].(*tls.AdminKubeConfigClientCertKey)
+	adminInternalClient := g[reflect.TypeOf(&kubeconfig.AdminInternalClient{})].(*kubeconfig.AdminInternalClient)
+
+	cfg := &tls.CertCfg{
+		Subject:      pkix.Name{CommonName: aroServiceName, Organization: []string{"system:masters"}},
+		KeyUsages:    x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
+		Validity:     tls.ValidityTenYears,
+	}
+
+	// new key and certificate will be stored in k.KeyRaw and k.CertRaw
+	err := clientCertKey.SignedCertKey.Generate(cfg, ca, "admin-kubeconfig-client", tls.DoNotAppendParent)
+	if err != nil {
+		return err
+	}
+
+	// create a Config for the new service kubeconfig based on the generated cluster admin Config
+	clusters := make([]clientcmd.NamedCluster, len(adminInternalClient.Config.Clusters))
+	copy(clusters, adminInternalClient.Config.Clusters)
```

Comment:
why is this needed? it looks like it's doing a shallow copy of a value which is not changed.